src/client/core/aad/auth-provider.ts

```diff
@@ -63,45 +79,80 @@ export default class AuthProvider {
                     { ...authRequest, prompt: "none" }
                 );
                 code = await authCodeCallback(url, true);
-            } catch (e) {
+            } catch (silentAuthExcpetion) {
                 url = await client.getAuthCodeUrl(authRequest);
                 code = await authCodeCallback(url);
             }
 
             const result: AuthorizationResult =
                 await client.acquireTokenByCode({ ...authRequest, code });
-            if (result) {
-                this._accounts[tenantId] = result.account;
-            }
             return result;
         }
     }
 
-    public logout(): void {
-        this._removeAccount();
+    public async logout(): Promise<void> {
+        this._logoutPromise = new Promise(resolve => {
```

Comment:
No need to create a new Promise object here. Also, this never actually awaits or returns the logout promise.

Suggestion:

```
public async logout(): Promise<void> {
    this._logoutPromise = this._removeAccounts();
    return this._logoutPromise;
}
```

Reply:
(or await this._logoutPromise for that last line - either way)